--- gallium/winsys/nouveau/drm/nouveau_drm_winsys.c	2014-10-03 05:59:51.000000000 +0200
+++ gallium/winsys/nouveau/drm/nouveau_drm_winsys.c	2015-07-02 21:33:30.494900651 +0200
@@ -120,7 +120,9 @@
 	if (!screen)
 		goto err;
 
-	util_hash_table_set(fd_tab, intptr_to_pointer(fd), screen);
+	//util_hash_table_set(fd_tab, intptr_to_pointer(fd), screen);
+    /* Use dupfd in hash table, to avoid crashes in ZaphodHeads configs */
+    util_hash_table_set(fd_tab, intptr_to_pointer(dupfd), screen);    
 	screen->refcount = 1;
 	pipe_mutex_unlock(nouveau_screen_mutex);
 	return &screen->base;
